### PR TITLE
add SkipBounds WriterOption to omit min/max bounds

### LINKED ISSUE #145: disable min/max page values per column
We're writing html files to parquet. Right now, page statistics are being written with the "min" and "max" values, but that isn't useful for a block of multiple megabytes of html, but it does bloat the file size.

There's a similar ticket for Java.
https://issues.apache.org/jira/browse/PARQUET-1911

My thought would be to add a new struct tag option, `nostats`, `nobounds`, `skipbounds` or something like that, and skip the min/max calculations. Would you be opposed to supporting that?

### PR DESCRIPTION
**Pull request by derekperkins**
Fixes #145

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -623,6 +624,15 @@ func SortingWriterConfig(options ...SortingOption) WriterOption {
 	return writerOption(func(config *WriterConfig) { config.Sorting.Apply(options...) })
 }
 
+// SkipBounds lists the path to a column that shouldn't have bounds written to the
+// footer of the parquet file. This is useful for data blobs, like a raw html file,
+// where the bounds are not meaningful.
+//
+// This option is additive, it may be used multiple times to skip multiple columns.
+func SkipBounds(path ...string) WriterOption {
```
Comment (achille-roussel) [inline config.go:632]: What do you think about naming this `SkipPageBounds` to be more specific and consistent with `SkipPageIndex`?

File: writer.go [hunk 2]
```diff
@@ -1150,6 +1160,7 @@ type writerColumn struct {
 	bufferIndex    int32
 	bufferSize     int32
 	writePageStats bool
+	skipBounds     bool
```
[the remark this answers is not in the document]
Comment (derekperkins) [reply writer.go:1163]: agreed

File: writer.go [hunk 3]
```diff
@@ -604,6 +604,7 @@ func newWriter(output io.Writer, config *WriterConfig) *writer {
 			bufferIndex:        int32(leaf.columnIndex),
 			bufferSize:         int32(float64(config.PageBufferSize) * 0.98),
 			writePageStats:     config.DataPageStatistics,
+			skipBounds:         shouldSkipBoundsColumn(config.SkipBounds, leaf.path),
```
[the remark this answers is not in the document]
Comment (derekperkins) [reply writer.go:607]: will do

File: config.go (hunk 1 above)
Comment (derekperkins) [reply config.go:632]: great

**Comment by derekperkins** [comment]
@achille-roussel thanks for the review. I made all the changes requested, and added a commit upgrading Go to v1.21, including bumping all dependencies to their latest versions. Let me know if that's overreaching.

**Comment by derekperkins** [comment]
Working on fixing that broken test

**Comment by derekperkins** [comment]
I inverted the slices func inside instead of outside. This should be good now

**Comment by derekperkins** [comment]
FWIW, I deployed this to test it out and it worked great

File: config.go [hunk 4]
```diff
@@ -213,6 +213,7 @@ type WriterConfig struct {
 	BloomFilters         []BloomFilterColumn
 	Compression          compress.Codec
 	Sorting              SortingConfig
+	SkipPageBounds       []columnPath
```
Comment (achille-roussel) [inline config.go:216]: We should use []string here or it wouldn't be possible to initialize the value because columnPath is unexported.

**Review by achille-roussel — APPROVED** [review]
I left one more minor change request, but it looks good otherwise!

File: config.go (hunk 4 above)
Comment (derekperkins) [reply config.go]: updated

**Comment by achille-roussel** [comment]
Thanks for your contribution 👏

**Pull request merged**